mypy/plugins/attrs.py

```diff
@@ -640,11 +640,13 @@ def _make_frozen(ctx: 'mypy.plugin.ClassDefContext', attributes: List[Attribute]
 def _add_init(ctx: 'mypy.plugin.ClassDefContext', attributes: List[Attribute],
               adder: 'MethodAdder') -> None:
     """Generate an __init__ method for the attributes and add it to the class."""
-    adder.add_method(
-        '__init__',
-        [attribute.argument(ctx) for attribute in attributes if attribute.init],
-        NoneType()
-    )
+    args = [attribute.argument(ctx) for attribute in attributes if attribute.init]
+    if all(getattr(a.variable.type, "type_of_any", None) == TypeOfAny.unannotated for a in args):
+        # This is a total hack , but makes --disallow-incomplete-defs usable with
```

Comment:
Maybe I wouldn't say hack. Maybe something about how this is a workaround but that there is a more correct solution.